internal/vds/axis.h

```diff
@@ -0,0 +1,38 @@
+#ifndef AXIS_H
+#define AXIS_H
+
+#include <memory>
+#include <string>
+
+#include <OpenVDS/OpenVDS.h>
+
+#include "vds.h"
+
+class Axis {
+public:
+    Axis(
+        enum axis_name const                    axis_name,
```

Comment:
I agree that we seem to disagree. 😄 

I am a bit torn about this class. I am a bit annoyed with some of the functionality, e.g., that it can map between the different dimensions/indices of the different coordinate systems... As you also say, this makes this class (too) knowledgeable of what is going on.

On the other hand the functionality makes it quite nice to use. Would you vote for splitting and move the conversion between request axes and VDS axis into its own class? Maybe this should be integrated in the VDS metadata/data handler? I am unsure about what component should be responsible for the conversions.